app/assets/src/components/views/PipelineViz.jsx

```diff
@@ -14,64 +14,65 @@ class PipelineViz extends React.Component {
   constructor(props) {
     super(props);
     this.pipelineVersion = this.props.stageResults.pipeline_version;
-    this.stageGraphsInfo = [];
+    this.stagesData = this.stagesDataWithModifiedStepNames();
     this.graphs = [];
+    this.graphContainers = [];
 
     this.stageNames = [
       "Host Filtering",
       "GSNAPL/RAPSEARCH alignment",
       "Post Processing",
+      ...(props.admin ? ["Experimental"] : []),
     ];
-    if (this.props.admin) {
-      this.stageNames.push("Experimental");
-    }
 
     this.state = {
       stagesOpened: [true, true, true, true],
       zoom: 1,
     };
-
-    this.handleMouseWheelZoom = this.handleMouseWheelZoom.bind(this);
   }
 
   componentDidMount() {
     this.drawGraphs();
   }
 
-  handleMouseWheelZoom(e) {
+  handleMouseWheelZoom = e => {
     const { zoomChangeInterval } = this.props;
     const zoomChange = (e.deltaY < 0 ? 1 : -1) * zoomChangeInterval;
     this.setState({ zoom: this.state.zoom + zoomChange });
-  }
+  };
 
   toggleStage(index) {
     const updatedStagesOpened = [...this.state.stagesOpened];
     updatedStagesOpened[index] = !updatedStagesOpened[index];
     this.setState({ stagesOpened: updatedStagesOpened });
   }
 
-  modifyStepNames() {
+  stagesDataWithModifiedStepNames() {
     // Strips 'PipelineStep[Run/Generate]' from front of each step name.
     // TODO(ezhong): Consider adding 'name' field to dag_json later.
     const { stageResults } = this.props;
-    forEach(stageResults.stages, (stageData, _) => {
-      stageData.steps.forEach(step => {
-        if (step.class.substring(0, 12) == "PipelineStep") {
-          step.class = step.class.substring(12);
+    const stagesWithModifiedNames = mapValues(stageData => {
+      const modifiedStageData = Object.assign({}, stageData);
+      modifiedStageData.steps = modifiedStageData.steps.map(step => {
+        const modifiedStep = Object.assign({}, step);
+        if (modifiedStep.class.substring(0, 12) == "PipelineStep") {
```

Comment:
I think a regex would be simpler here
```js
return {
  ...step,
  class: step.replace(/(PipelineStep|Run|Generate)/, '')
}
```
I'm also replacing the `Object.assign` and the `return` here as well.
NOTE: this assumes the intended behavior was for the `if`s to be mutually exclusive.

It would require some modification if you wanted to do a transform like `PipelineStepRunGeneratehello` -> `hello`